scalableminds/webknossos
# Split cells via Min Cut

This PR:
- implements a min-cut approach to cut a cell given two seeds
- fixes a rendering bug where fallback rendering did not use the best available mag sometimes

### URL of deployed dev instance (used for testing):
- https://mincut.webknossos.xyz

### Steps to test:
- open a volume tracing with fallback layer
- use adhoc meshing for a cell to find a good spot to cut (e.g., a spinehead)
- create a tree place two connected nodes in the cell which represent the seeds (e.g., one in the dendrite and one in the spinehead)
- rightclick the tree and select "perform min-cut"
- a min cut should be performed which erases some voxels between the two seeds
- you can verify that the split was successful by using the floodfill with a new segment id on one half of the segment

### To do:
- [X] implement working version
- [X] clean up UI
- [x] pull master
- [x] generalize mag handling (currently, the existence of 2-2-1 is assumed)
- [x] try to refine min-cut on better mags
- [x] make operation undo-able
- ~[ ] fix flicker bug when zooming from mag 4 to mag 8.~ extracted to https://github.com/scalableminds/webknossos/pull/6029

### Issues:
- fixes #5934 

------
(Please delete unneded items, merge only when none are left open)
